src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +67,41 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
         return $eventRecord;
     }
 
-    public function add(EventRecordInterface $eventRecord): void
+    public function addAndCommit(EventRecordInterface $eventRecord): void
     {
-        $this->entityManager->persist($eventRecord);
-    }
+        // use persister to insert only given entity instead of flushing all managed entities via the entity manager
+        // this prevents flushing unrelated changes and allows to call this method in a postFlush event-listener
 
-    public function commit(): void
-    {
-        $this->entityManager->flush();
+        $userId = null;
+        if ($eventRecord->getUser()) {
+            $userId = $eventRecord->getUser()->getId();
+        }
+
+        $queryBuilder = $this->entityManager->getConnection()->createQueryBuilder();
+        $queryBuilder
+            ->insert('el_event_records')
+            ->values(
+                [
+                    'eventType' => $queryBuilder->createNamedParameter($eventRecord->getEventType()),
+                    'eventContext' => $queryBuilder->createNamedParameter(\json_encode($eventRecord->getEventContext(), true)),
+                    'eventDateTime' => $queryBuilder->createNamedParameter($eventRecord->getEventDateTime()->format('Y-m-d H:i:s')),
+                    'eventBatch' => $queryBuilder->createNamedParameter($eventRecord->getEventBatch()),
+                    'resourceKey' => $queryBuilder->createNamedParameter($eventRecord->getResourceKey()),
+                    'resourceId' => $queryBuilder->createNamedParameter($eventRecord->getResourceId()),
+                    'resourceLocale' => $queryBuilder->createNamedParameter($eventRecord->getResourceLocale()),
+                    'resourceWebspaceKey' => $queryBuilder->createNamedParameter($eventRecord->getResourceWebspaceKey()),
+                    'resourceTitle' => $queryBuilder->createNamedParameter($eventRecord->getResourceTitle()),
+                    'resourceTitleLocale' => $queryBuilder->createNamedParameter($eventRecord->getResourceTitleLocale()),
+                    'resourceSecurityContext' => $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityContext()),
+                    'resourceSecurityObjectType' => $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityObjectType()),
+                    'resourceSecurityObjectId' => $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityObjectId()),
+                    'userId' => $queryBuilder->createNamedParameter($userId),
+                ]);
+
+        if ($this->shouldPersistPayload) {
```

Comment:
would be nice to test this somewhere

Reply:
I honestly have no idea how I can test this properly

A custom config for one bundle is not working, because the database purge only truncates the tables and doesn't recreate them with the `payload` column. 

Do you have any ideas? @alexander-schranz @wachterjohannes

Reply:
i am open to always adding the `payload` column and only set a value if the `shouldPersistPayload` is true 🙂

Reply:
we need a second environment to test it like we did in other bundles where we tested different configuration or extending of entities.

Reply:
not sure if this is worth the effort - lets discuss this offline 🙂